### PR TITLE
Feat/201/create an application form and allow sign in for non wizeline email users

### PR DESCRIPTION
**Pull request by AxelUZ**
# What does this PR do?
Create an application form to fill out the Applicants table information

# Where should the reviewer start?
/app/routes/applicationForm/$applicantId/index.tsx
/app/routes/applicationForm/$applicantId/createapplicant.tsx

# How should this be manually tested?
1. Go to /applicationForm/"YourProfileId"
2. Fill in the form fields and submit it
3. You will be redirected to /applicants
4. Check if your information is there

# Any background context you want to provide?

# What are the relevant tickets?
#201 

# Screenshots
Captura de pantalla 2023-09-04 a la(s) 10 41 06 p m
Captura de pantalla 2023-09-04 a la(s) 10 41 16 p m
Captura de pantalla 2023-09-04 a la(s) 10 41 30 p m

# Questions

# Checklist

- [ ] I added the necessary documentation, if appropriate.
- [ ] I added tests to prove that my fix is effective or my feature works.
- [ ] I reviewed existing Pull Requests before submitting mine.

### CODE REVIEW

**Review by jackbravo — COMMENTED** [review]
This is a first shallow review 😋

File: app/routes/applicationForm/$applicantId/index.tsx [hunk 1]
```diff
@@ -0,0 +1,397 @@
...
+    wizelinePrograms: z.string().optional(),
+    comments: z.string().optional(),
+  })
+);
+  
+  export default function FormPage() {
+    return (
+    <Container>
+      <img src='/HeaderImage.png' alt='Wizeline' style={{width:"100%"}} />
+      <h1>Application Form</h1>
```
Comment (martinrobled0) [inline app/routes/applicationForm/$applicantId/index.tsx:52]: You can replace the html like this:
`<Typography component="div"> Application For</Typography>`

File: prisma/seed.ts [hunk 2]
```diff
@@ -555,6 +555,17 @@ async function seed() {
       department: "Engineering",
     },
   });
+  await db.profiles.upsert({
+    where: { email: "	axel.uzetagomez@in.wizeline.com" },
```
Comment (martinrobled0) [inline prisma/seed.ts:559]: delete the space here

**Review by martinrobled0 — CHANGES_REQUESTED** [review]

File: app/routes/applicationForm/$applicantId/index.tsx [hunk 3]
```diff
@@ -0,0 +1,397 @@
...
+    interestedRoles: z.string().optional(),
+    preferredTools: z.string().optional(),
+    startDate: z.string().min(1,{message: "This field is Required"}), 
+    endDate: z.string().min(1,{message: "This field is Required"}), 
+    hoursPerWeek: z.string().min(1,{message: "This field is Required"}), 
+    howDidYouHearAboutUs: z.string().min(1,{message: "This field is Required"}),
+    participatedAtWizeline: z.string().min(1,{message: "This field is Required"}),
+    wizelinePrograms: z.string().optional(),
+    comments: z.string().optional(),
+  })
```
Comment (CarlosMilano) [inline app/routes/applicationForm/$applicantId/index.tsx:45]: Add more validation for fields to avoid incorrect data

**Review by jackbravo — APPROVED** [review]
The direction of this looks good to me. So as soon as you address most comments you can commit 😊

File: package.json [hunk 4]
```diff
@@ -31,8 +31,11 @@
     "@emotion/styled": "^11.8.1",
     "@google-cloud/bigquery": "^6.0.3",
     "@mui/icons-material": "^5.8.4",
+    "@mui/lab": "^5.0.0-alpha.141",
```
[the remark this answers is not in the document]
Comment (AxelUZ) [reply package.json]: I forgot to remove it 😬

File: package.json [hunk 5]
```diff
@@ -31,8 +31,11 @@
...
     "@mui/material": "^5.8.3",
     "@mui/x-data-grid": "^5.17.26",
+    "@mui/x-data-grid-generator": "^6.11.2",
+    "@mui/x-date-pickers": "^6.11.2",
```
[the remark this answers is not in the document]
Comment (AxelUZ) [reply package.json]: yes, I will remove it

File: prisma/seed.ts [hunk 6]
```diff
@@ -555,6 +555,16 @@ async function seed() {
...
     },
   });
+  await db.profiles.upsert({
+    where: { email: "axel.uzetagomez@in.wizeline.com" },
+    update: {},
+    create: {
+      email: "axel.uzetagomez@in.wizeline.com",
+      firstName: "Axel",
+      preferredName: "Axel",
+      lastName: "Uzeta",
```
Comment (martinrobled0) [inline prisma/seed.ts:565]: I think that it's missing the department field

Comment (AxelUZ) [reply prisma/seed.ts:565]: It's done 👍

**Review by martinrobled0 — APPROVED** [review]
good job!

**Comment by AxelUZ** [comment]
> good job!
Thx  🙏

**Pull request merged**